mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/MapperBuilderGenerator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor;
+
+import com.datastax.oss.driver.api.core.session.Session;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeSpec;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+
+public class MapperBuilderGenerator extends FileGenerator {
+
```

Comment:
Will client use this API in the following way:
`
Session session = initSession();
InventoryMapperBuilder.build(session).getProductDao()
`?
Why we are generating also `build()` (parameterless) method? How `session` is initialized in that case?

Reply:
The session is mandatory so I made it a constructor argument of the builder:
```java
InventoryMapper mapper = new InventoryMapperBuilder(session)
  // .withXxx(...)
  // .withYyy(...)
  .build();
```
My instinct tells me that there will be other optional parameters that you can set on the builder. But if that turns out to not be true, we can revisit the pattern.